frontend/src/routes/index.lazy.tsx

```diff
@@ -0,0 +1,13 @@
+import { createLazyFileRoute } from "@tanstack/react-router";
+
+export const Route = createLazyFileRoute("/")({
+  component: Index,
+});
+
+function Index() {
+  return (
+    <div className="p-2">
+      <h3>Welcome Home!</h3>
+    </div>
+  );
+}
```

Comment:
If the user isn't logged in, it should redirect them to the login form, you can take the code in `App.tsx` and after the query has finished, you can check if the user object is provided or not, if it is provided, you redirect to login, else, you show the welcome page with their usename for ex.